httpapi_plugins/ftd.py

```diff
@@ -129,6 +129,31 @@ def refresh_token_payload(refresh_token):
             raise ConnectionError(
                 'Server returned response without token info during connection authentication: %s' % response)
 
+    def _lookup_login_url(self, payload):
+        """When token generation URL is not defined in the inventory we may try to find it on the fly.
+        :param payload: Token request payload
+        :type payload: dict
+        :return: token generation response
+        """
+        for url in self._get_known_token_paths():
+            try:
+                response = self._login_api_call(payload, url)
+            except ConnectionError:
+                display.vvvv('REST:request to {0} failed because of connection error'.format(url))
+            else:
+                self._set_api_token_path(url)
+                return response
+
+        raise ConnectionError("Token generation API was not found.")
+
+    def _login_api_call(self, payload, url):
```

Comment:
Please, rename this method to start with a verb.

Reply:
fixed